ElevatorSim/src/elevatorsim/elevator/ElevatorServer.java

```diff
@@ -39,23 +52,199 @@ public DatagramPacket handleElevatorRequest(DatagramPacket request) {
+				timerTask, 
+		        TimeConstants.changeDoorState );
+			} else if (newState == ElevatorState.MOTOR_UP) {
+				elevator.setElevatorState(ElevatorState.MOTOR_UP);
+				timer = new Timer();
+				timerTask = new TimerTask() {
+		            @Override
+		            public void run() {
+		            	if(elevator.getElevatorState() == ElevatorState.MOTOR_UP) {
+		            		elevator.setFloor(elevator.getFloor()+1);
+		            		try {
+								sender.send( MessagePackets.generateElevatorEvent(new ElevatorEvent(elevator.getFloor())),InetAddress.getByName(NetworkConstants.SCHEDULER_IP), NetworkConstants.SCHEDULER_PORT);
+							} catch (UnknownHostException e) {
+								// TODO Auto-generated catch block
+								e.printStackTrace();
+							} catch (IOException e) {
+								// TODO Auto-generated catch block
+								e.printStackTrace();
+							}
+		            	}else {
+		            		timer.cancel();
+		            	}
+		            }
+		        };
+				timer.scheduleAtFixedRate( 
+				timerTask, 
+				TimeConstants.moveOneFloor,TimeConstants.moveOneFloor );
+				
+				
+			} else if(newState == ElevatorState.MOTOR_DOWN) {
+				
+				elevator.setElevatorState(ElevatorState.MOTOR_DOWN);
+				timer = new Timer();
+				timerTask = new TimerTask() {
+		            @Override
+		            public void run() {
+		            	if(elevator.getElevatorState() == ElevatorState.MOTOR_DOWN) {
+		            		elevator.setFloor(elevator.getFloor()-1);
+		            		try {
+								sender.send( MessagePackets.generateElevatorEvent(new ElevatorEvent(elevator.getFloor())),InetAddress.getByName(NetworkConstants.SCHEDULER_IP), NetworkConstants.SCHEDULER_PORT);
+							} catch (UnknownHostException e) {
+								// TODO Auto-generated catch block
+								e.printStackTrace();
+							} catch (IOException e) {
+								// TODO Auto-generated catch block
+								e.printStackTrace();
+							}
+		            	}else {
+		            		timer.cancel();
+		            	}
+		            }
+		        };
+				timer.scheduleAtFixedRate( 
+				timerTask, 
+		        TimeConstants.moveOneFloor,TimeConstants.moveOneFloor );
+				
+			}else if (newState != ElevatorState.DOOR_CLOSED) {
+				error = true;
+			}
+		}else if (currentState == ElevatorState.MOTOR_UP || currentState == ElevatorState.MOTOR_DOWN) {
+			if(newState == ElevatorState.DOOR_CLOSED) {
+				this.elevator.setElevatorState(ElevatorState.DOOR_CLOSED);
+				try {
+					sender.send(MessagePackets.generateElevatorStateChange(new ElevatorStateChange(ElevatorState.DOOR_CLOSED)),InetAddress.getByName(NetworkConstants.SCHEDULER_IP), NetworkConstants.SCHEDULER_PORT);
+				} catch (UnknownHostException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				}
+			} else if (newState != currentState) {
+				error = true;
+			}
+		}
+		
+		return error ? MessagePackets.Responses.RESPONSE_FAILURE() : MessagePackets.Responses.RESPONSE_SUCCESS();
+		
 	}
 
+	//@Override
```

Comment:
remove this code if it isn't needed anymore

Reply:
removed